core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1254,23 +1254,7 @@ class KafkaController(val config: KafkaConfig,
       if (!isActive) {
         0
       } else {
-        controllerContext.allPartitions.count { topicPartition =>
```

Comment:
Ouch. So the main issue is that we were doing this full iteration overall partitions on _every_ controller event.

Reply:
Yeah...